### ISSUE TITLE
Repeat the upload

### ISSUE THREAD

**Issue by lixiangyue** [nervgh/angular-file-upload#813]
Why can't upload the same file？
then there's a problem！
first， upload a file
second， remove the file
third， can't upload files the same as the first file.

**Comment by lixiangyue** [comment]
Why can't select the same file？
then there's a problem！
first， select a file
second， select the file
third， can't select files the same as the first file.
不能选择同一份文件的
不然会造成一个问题
1、选择一份文件
2、删除那份文件
3、再选择之前的那份文件
4、最终结果表格队列没有加到文件

**Comment by PaulDMendoza** [comment]
I have the same issue.

**Comment by stevebelt** [comment]
Same issue...
More exactly, I cannot upload the same file twice in a row: 
(I have set autoUpload: true, removeAfterUpload: true)

Now:
1. Select File_A.txt -> success
2. Select File_A.txt -> nothing happens; onWhenAddingFileFailed NOT invoked.
3. Select File_B.txt -> success
4. Select File_A.txt -> success

**Comment by PaulDMendoza** [comment]
Yeah, same issue. Can't upload the same file twice in a row. Very strange.

**Comment by stevebelt** [comment]
I found a solution on the Net.  I added this line of code and now I do not get this error.

FileUploader.FileSelect.prototype.isEmptyAfterSelection = function() {
        return true; // true|false
    };
 This is based upon the following FAQ:

https://github.com/nervgh/angular-file-upload/wiki/FAQ#4-no-file-chosen-or-re-add-same-file
